### ISSUE TITLE
"Feed to listbox" button is gone

### ISSUE THREAD

**Issue by Andreas0602** [doublecmd/doublecmd#621]
On macOS, starting with snapshot version doublecmd-1.1.0-10394.cocoa.x86_64, in the results of a file search, it is no longer possible to feed the search results into the listbox. In previous versions, there was a button for that. That button is gone now. Please bring it back.

**Comment by alexx2000** [comment]
I guess it is a Lazarus 2.3 bug which used for snapshots now.

**Comment by rich2014** [comment]
> I guess it is a Lazarus 2.3 bug which used for snapshots now.

no, the button was gone about a month ago.

**Comment by alexx2000** [comment]
So some previous Lazarus update had broken it. DC file search window code did not change at last 4 month. Buttons disappear on other tabs too.

**Comment by rich2014** [comment]
the cause has been found and a PR has been submitted #626

**Comment by rich2014** [comment]
are there any other Forms with similar problems?

**Comment by Skif-off** [comment]
`AutoSize = True` adds a problem in Linux (main tab "Standard", I have to resize the window in every session),
Maybe it's better to report it to the Lazarus's bug tracker?

P.S. Lazarus 2.2.4 and 2.3.0 (from 'main').

**Comment by Skif-off** [comment]
Up, 1.0.8 vs. 1.1.0:

![findfiles](https://user-images.githubusercontent.com/6333198/204196301-cd41f216-9b01-40ac-8fc7-f23f3296b9b2.png)

**Comment by alexx2000** [comment]
Yes, also `AutoSize = True` broke saving/restoring window size. So it's need to find other solution.

**Comment by rich2014** [comment]
I rechecked the code of TCustomForm and found that it works fine in Lazarus 2.2, just because of side effect of TCustomForm.getMonitor().

`Monitor.PixelsPerInch` is called in TCustomForm.create(), and SetBounds() is called in `TCustomForm.getMonitor()` in Lazarus 2.2.

but SetBounds() is NOT in called TCustomForm.getMonitor() in Lazarus 2.3.

but the height of the form should not depend on whether SetBounds() will be called in `TCustomForm.getMonitor()`.

i'll try to find the problem with calculating the height of FindDlg on Cocoa.

```pascal
procedure TCustomForm.AfterConstruction;
var
  MonPPI: Integer;
begin
  SetRestoredBounds(Left, Top, Width, Height, True);
  DoCreate;
  EndFormUpdate; // the BeginFormUpdate is in CreateNew
  inherited AfterConstruction;

  MonPPI := Monitor.PixelsPerInch;
  if Application.Scaled and Scaled and (MonPPI > 0) and (MonPPI <> PixelsPerInch)
  and not (csDesigning in ComponentState) then
    AutoAdjustLayout(lapAutoAdjustForDPI, PixelsPerInch, MonPPI,
                     Width, MulDiv(Width, MonPPI, PixelsPerInch));
end;
```

**Comment by rich2014** [comment]
the root cause found, fix by #717
